### PR TITLE
feat(plugin): SearchReply

### PR DESCRIPTION
**Pull request by Syncxv**
simple plugin that adds a reply button to search results

before:
before

after:
after

### CODE REVIEW

**Comment by Vendicated** [comment]
wait for context menu api

**Comment by Vendicated** [comment]
Context menu api is here now so you can use it instead!

**Comment by Syncxv** [comment]
ready

File: src/plugins/searchReply.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import { LazyComponent } from "@utils/misc";
+import definePlugin from "@utils/types";
+import { findByCode, findByCodeLazy, findLazy } from "@webpack";
+import { ChannelStore, Menu, SelectedChannelStore } from "@webpack/common";
+import { Message } from "discord-types/general";
+
+const ReplyIcon = LazyComponent(() => findByCode("M10 8.26667V4L3 11.4667L10 18.9333V14.56C15 14.56 18.5 16.2667 21 20C20 14.6667 17 9.33333 10 8.26667Z"));
+
+// for people who speak different languages
+const i18n = findLazy(m => m.Messages?.MESSAGE_ACTION_REPLY);
```
Comment (Vendicated) [inline src/plugins/searchReply.tsx:30]: Could you move this to Webpack commons please? I feel like many more plugins would want to use it, and I'm pretty sure MessageLogegr already does!

File: src/plugins/searchReply.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+const replyFn = findByCodeLazy("showMentionToggle", "TEXTAREA_FOCUS", "shiftKey");
+
+const messageContextMenuPatch: NavContextMenuPatchCallback = (children, props) => {
+    if (!props) return;
```
Comment (Vendicated) [inline src/plugins/searchReply.tsx:35]: i don't think this is a relevant case to check?

File: src/plugins/searchReply.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    const { message } = props as { message: Message; };
```
Comment (Vendicated) [inline src/plugins/searchReply.tsx:36]: you could just type the props argument up above I'm fairly certain

File: src/plugins/searchReply.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+    // make sure they are in the same channel as the message
+    if (!message || SelectedChannelStore.getChannelId() !== (message.channel_id ?? message.getChannelId())) return;
```
Comment (Vendicated) [inline src/plugins/searchReply.tsx:39]:
```suggestion
    if (!message || SelectedChannelStore.getChannelId() !== message.channel_id)) return;
```

File: src/plugins/searchReply.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+    const channel = ChannelStore.getChannel(message.channel_id ?? message.getChannelId());
```
Comment (Vendicated) [inline src/plugins/searchReply.tsx:41]: same as above!

**Review by Vendicated — CHANGES_REQUESTED** [review]
Looks good other than the feedback!

**Review by Vendicated — APPROVED** [review]

**Pull request merged**